### PR TITLE
feat: block body ✨

### PR DESCRIPTION
**Pull request by EvolveArt**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: #234 

# What is the new behavior?

- Adds a new `TransactionReceiptWrapper` type with `TryInto` trait implemented
- Get transaction receipt on `Invoke` transactions
- Adds TransactionReceipt field in `Pending` storage value
- Adds `transactions` field to `Block` structure

### CODE REVIEW

File: crates/pallets/starknet/src/lib.rs [hunk 1]
```diff
@@ -395,9 +391,16 @@ pub mod pallet {
...
+            let receipt;
             match call_info {
                 Ok(Some(mut v)) => {
-                    Self::emit_events(&mut v, &mut transaction).map_err(|_| Error::<T>::EmitEventError)?;
+                    let events = Self::emit_events(&mut v, &mut transaction).map_err(|_| Error::<T>::EmitEventError)?;
+                    receipt = TransactionReceiptWrapper {
+                        events: BoundedVec::try_from(events).unwrap(),
+                        transaction_hash: transaction.hash,
+                        tx_type: TxType::InvokeTx,
+                        actual_fee: U256::zero(), // TODO: switch to actual fee
```
Comment (abdelhamidbakhta) [inline crates/pallets/starknet/src/lib.rs:402]: please create issue and link here

File: crates/primitives/starknet/src/block/mod.rs [hunk 2]
```diff
@@ -1,11 +1,43 @@
...
 pub use header::*;
+use sp_core::{ConstU32, H256};
+
+use crate::transaction::types::Transaction;
 
 /// Serializer
 pub mod serialize;
 
+/// Block transactions max size
+pub type MaxTransactions = ConstU32<4294967295>; // TODO: add real value
```
Comment (abdelhamidbakhta) [inline crates/primitives/starknet/src/block/mod.rs:14]: add issue and link

File: crates/primitives/starknet/src/block/mod.rs [hunk 3]
```diff
@@ -22,12 +54,14 @@ pub mod serialize;
...
     header: Header,
+    /// The block transactions.
+    transactions: BlockTransactions,
 }
 
 impl Block {
     /// Creates a new block.
-    pub fn new(header: Header) -> Self {
-        Self { header }
+    pub fn new(header: Header, transactions: BlockTransactions) -> Self {
```
Comment (abdelhamidbakhta) [inline crates/primitives/starknet/src/block/mod.rs:63]: update rust doc with arguments

File: crates/primitives/starknet/src/transaction/mod.rs [hunk 4]
```diff
@@ -141,6 +142,36 @@ impl Default for EventWrapper {
     }
 }
 
+/// Try to convert a `&TransactionReceipt` into a `TransactionReceiptWrapper`.
+impl TryInto<TransactionReceiptWrapper> for &TransactionReceipt {
+    type Error = EventError;
+
+    /// TODO: add block hash and block number
```
Comment (abdelhamidbakhta) [inline crates/primitives/starknet/src/transaction/mod.rs:149]: add issue and link
add rust doc

File: crates/primitives/starknet/src/transaction/types.rs [hunk 5]
```diff
@@ -5,7 +5,8 @@ use starknet_api::StarknetApiError;
 
 use crate::execution::{CallEntryPointWrapper, ContractAddressWrapper, ContractClassWrapper};
 
-/// Max size of the event array.
+/// Max size of arrays.
+/// TODO: add real value
 pub type MaxArraySize = ConstU32<4294967295>;
```
Comment (abdelhamidbakhta) [inline crates/primitives/starknet/src/transaction/types.rs:10]: add issue and link

File: crates/primitives/starknet/src/transaction/mod.rs [hunk 6]
```diff
@@ -141,6 +142,36 @@ impl Default for EventWrapper {
...
+    fn try_into(self) -> Result<TransactionReceiptWrapper, Self::Error> {
```
Comment (abdelhamidbakhta) [inline crates/primitives/starknet/src/transaction/mod.rs:150]: add unit tests for success and error cases

File: crates/pallets/starknet/src/lib.rs [hunk 7]
```diff
@@ -721,24 +725,29 @@ pub mod pallet {
...
-                sequencer_address,
-                block_timestamp,
-                transaction_count,
-                transaction_commitment,
-                event_count,
-                event_commitment,
-                protocol_version,
-                extra_data,
-            ));
+            let block = StarknetBlock::new(
```
Comment (abdelhamidbakhta) [inline crates/pallets/starknet/src/lib.rs:735]: add new test / update existing tests to cover the change of this PR, i.e check presence of transactions as well as receipt, etc

**Review by abdelhamidbakhta — CHANGES_REQUESTED** [review]
left some comments

File: crates/pallets/starknet/src/lib.rs (hunk 1 above)
Comment (EvolveArt) [reply crates/pallets/starknet/src/lib.rs:402]: https://github.com/keep-starknet-strange/madara/issues/251

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm

**Pull request merged**